### ISSUE TITLE
Kernel current working directory when file is moved

### ISSUE THREAD

**Issue by jasongrout** [jupyterlab/jupyterlab#5989]
If we move an .ipynb file with a running kernel, it seems that the kernel's current working directory isn't changed. I think this makes sense thinking about the kernel process as a separate thing from the file location, but users can be confused.

What is our story around this workflow? Should we offer to restart a kernel to make it easier on user expectations? Should we just say this is working as designed?

**Comment by bollwyvl** [comment]
I don't see how you could generally maintain open file handles, etc. If
anything, introduce a UI element that indicates Something Is Not Right,
like /!\ next to/in place of the kernel indicator.

**Comment by jasongrout** [comment]
I agree that we can't really affect the kernel (I mean, who knows how to change the current working directory of random kernel XYZ?). I think the best we'd be able to do is offer to restart the kernel with the "right" working directory, or like you said, just flag the potential problem for the user.

**Comment by ellisonbg** [comment]
Another things to consider - there is no guarantee the kernel is running on the same system and seeing the same files as the UI. So even restarting the kernel in that dir may not be possible.

**Comment by n-a-sz** [comment]
Maybe a warning message when the notebook is moved and there is a running kernel for it would be be enough. Also the message could contain that a brand new python kernel is needed and restarting it won't solve the working directory issue (if it is a feature of restarting the kernel and not a bug).

**Comment by MMesch** [comment]
The triaging agent has recommended to merge this issue with #3856 which makes sense to me. Accordingly, I'm closing this one and suggest continuing the discussion there. Please reopen and comment why if you think this issue should remain open separately.
